packages/discordx/src/decorators/classes/DSimpleCommand.ts

```diff
@@ -203,79 +203,89 @@ export class DSimpleCommand extends Method {
-          return Number(args[index]);
-        }
-
-        // Channel | undefined
-        if (op.type === "CHANNEL") {
-          if (!id?.length) {
-            return undefined;
-          }
-          return command.message.guild?.channels.resolve(id);
-        }
+    return Promise.all(
+      this.options
+        .sort((a, b) => (a.index ?? 0) - (b.index ?? 0))
+        .map((op, index) => {
+          // only digits
+          const id = args[index]?.replace(/\D/g, "");
+          const invalidError = Error(`Invalid id given: ${args[index]}`);
 
-        // Role | undefined
-        if (op.type === "ROLE") {
-          if (!id?.length) {
+          // undefined
+          if (!args[index]?.length) {
             return undefined;
           }
 
-          return command.message.guild?.roles.resolve(id);
-        }
+          // Boolean
+          if (op.type === "BOOLEAN") {
+            return Boolean(args[index]);
+          }
 
-        // GuildMember | User | undefined
-        if (op.type === "USER") {
-          if (!id?.length) {
-            return undefined;
+          // Number
+          if (op.type === "NUMBER" || op.type === "INTEGER") {
+            return Number(args[index]);
           }
 
-          if (command.message.channel.type === "DM") {
-            return command.message.client.user?.id === id
-              ? command.message.client.users.resolve(id)
-              : command.message.author;
+          // Channel | undefined
+          if (op.type === "CHANNEL") {
+            if (!id?.length || id.length < 16 || id.length > 20) {
+              return invalidError;
+            }
+
+            return command.message.guild?.channels
+              .fetch(id)
+              .catch((err) => err);
           }
 
-          return command.message.guild?.members.resolve(id);
-        }
+          // Role | undefined
+          if (op.type === "ROLE") {
+            if (!id?.length || id.length < 16 || id.length > 20) {
+              return invalidError;
+            }
 
-        // GuildMember | User | Role | undefined
-        if (op.type === "MENTIONABLE") {
-          if (!id?.length) {
-            return undefined;
+            return command.message.guild?.roles.fetch(id).catch((err) => err);
           }
 
-          if (command.message.channel.type === "DM") {
-            return command.message.client.user?.id === id
-              ? command.message.client.users.resolve(id)
-              : command.message.author;
+          // GuildMember | User | undefined
+          if (op.type === "USER") {
+            if (!id?.length || id.length < 16 || id.length > 20) {
+              return invalidError;
+            }
+
+            if (command.message.channel.type === "DM") {
+              return command.message.client.user?.id === id
+                ? command.message.client.user
+                : command.message.author.id === id
+                ? command.message.author
+                : undefined;
```

Comment:
You should return an error here as well, otherwise it will look like the user didn't give an argument here, when in reality they gave an incorrect one

Reply:
> You should return an error here as well, otherwise it will look like the user didn't give an argument here, when in reality they gave an incorrect one

agree, not throwing an error and returning undefined hides any issues that the user will have;